src/modules/airspeed_selector/airspeed_selector_main.cpp

```diff
@@ -303,9 +306,9 @@ AirspeedModule::Run()
 		bool fixed_wing = !_vtol_vehicle_status.vtol_in_rw_mode;
 		bool in_air = !_vehicle_land_detected.landed;
 
-		/* Prepare data for airspeed_validator */
-		struct airspeed_validator_update_data input_data = {};
-		input_data.timestamp = _time_now_usec;
+		// Prepare data for airspeed_validator
+		airspeed_validator_update_data input_data{};
+		input_data.timestamp = _vehicle_local_position.timestamp_sample;
```

Comment:
It's minor, but it minimizes most of the jitter in your delta time (the only use I saw at a glance) using the time that corresponds with the input data rather than when the module happened to actually run.